File: algorithms/fibonacci/fib.py
import sys

def fib_it(a):
  # iterative version
  ret = 0
  stk = [a]
  while len(stk) > 0:
    a = stk.pop()  
    if a == 0:
      continue
    elif a == 1:
      ret += 1
    else:
      stk.append(a-1)
      stk.append(a-2)
  return ret
    
def fib_rec(a):
  # recursive version
  if a == 0:
    return 0
  elif a == 1:
    return 1
  return fib(a-1) + fib(a-2)

n = int(sys.argv[1])
print(fib_it(n))

File: algorithms/fibonacci/fib.cpp
#include <iostream>

int fib(int n){
  if (n == 0) return 0;
  if (n == 1) return 1;
  return fib(n-1) + fib(n-2);
}

int main(){
  int a[] = {30, 20, 21, 10, 15, 0, 5, 1};
  int len = sizeof(a)/sizeof(int);
  for (int i=0; i<len; i++) {
    std::cout << fib(a[i]) << " ";
  }
  std::cout << "\n";
}
